projects/WTI-UI/src/app/modules/runs/components/new-run/new-run.component.ts

```diff
@@ -196,6 +199,34 @@ export class NewRunComponent implements OnInit, OnDestroy {
 	//none of the specified FileSubmission files contains a space in its name
 	return false;
 	}
+
+  //returns true if there are duplicate filenames among any of the specified FileSubmissions; false if all unique.
+  private filenameContainsDuplicates(mainfile: FileSubmission, additionalFiles: FileSubmission []): boolean {
+    let mainfilename = mainfile.fileName;
```

Comment:
I'm not the typescript expert, but you may wish to check if typescript has a similar thing regarding file separators for pathname components..

Reply:
I haven't tried it myself, but a quick search led me to [this page](https://dirask.com/posts/TypeScript-Node-js-get-platform-specific-path-segment-separator-pJAboD), which says that the Typescript equivalent to Java's `File.separator` property is the following:
```typescript
import * as path from 'path';
```
and then reference the property `path.sep` to get the platform-specific file separator.  You can also do things like
```typescript
filePath.split(path.sep)
```
to separate a full path into its components based on the value of `path.sep`.

I think this should be put to use in the changes to new-run.component.ts.

Reply:
To use this, we need to install the package `npm install --save @types/node`. It can be added to the `package.json`. Without it the build fails.

Reply:
So, I guess you are asking if it's OK to add that to the packages that **node** installs.  Did you actually try what you are suggesting and confirm it works?  If so, then I don't see a problem with adding it unless @clevengr does.

Reply:
@types/node is already present in the `package.json`. Need to check why it still fails.

Reply:
It is already included in the package.json. I tried a few different things like moving the @types/node to dependencies from devDependencies in package.json, adding "types": ["node"] in compilerOptions of tsconfig.json but everytime got error TS2307: Cannot find module 'path' or its corresponding type declarations.